vscode_extension/src/editor/definitions/definitionVisualEditor.js

```diff
@@ -0,0 +1,109 @@
+/* tslint:disable */
+/* eslint-disable */
+const vscode = acquireVsCodeApi();
+
+// This script is run within the webview itself
+(function () {
+    class DefinitionEditor {
+        constructor( /** @type {HTMLElement} */ parent) {
+            this.ready = false;
+
+            this.editable = false;
+            this.container = parent;
+            this.rootKey = "";
+            this.sourceUri = "";
+            this.editor = null;
+        }
+
+        async setEditorData(data) {
+            // Initialize the editor with a JSON schema
+            this.rootKey = Object.keys(data.structure)[0];
+            addTitlesToJsonSchema(data.jsonSchema, this.rootKey)
+
+            if (this.editor) {
+                this.editor.destroy();
+            }
+
+            this.editor = new JSONEditor(document.getElementById('main'),
+                {
+                    "use_default_values": true,
+                    "prompt_before_delete": false,
+                    "disable_collapse": false,
+                    "disable_array_delete_last_row": true,
+                    "array_controls_top": true,
+                    "disable_edit_json": true,
+                    "schema": data.jsonSchema,
+                    "startval": data.structure[this.rootKey]
+                }
+            );
+
+            this.editor.on('change', () => {
+                const editorContent = this.editor?.getValue()
+                const rootKey = editor.rootKey
+                const definitionStructure = {}
+                definitionStructure[rootKey] = editorContent
+
+                // Enum AacEditorEventTypes.EDIT = 2
+                const aacDefinitionEdit = {
+                    name: definitionStructure[rootKey].name,
+                    sourceUri: this.sourceUri,
+                    structure: definitionStructure
+                };
+
+                vscode.postMessage({ type: 2, body: aacDefinitionEdit});
+            });
+
+            // Hook up the submit button to log to the console
+            document.getElementById('submit').addEventListener('click', function () {
+                // Enum AacEditorEventTypes.SAVE = 3
+                vscode.postMessage({ type: 3 });
```

Comment:
Is there a way to import the `AacEditorEventTypes` enum here? If not, maybe we redefine it above so that references to the enum are more readable without comments. It'd also remove the need to find everywhere that 1, 2, 3, etc is used if/when things change.

Reply:
I don't think that we can import it because of the separation between the extension's code and this javascript file that's run inside an embedded frame. That's a good call on creating an enum in the JS file, though.